backend/models/customer.js

```diff
@@ -3,15 +3,42 @@ const ClientBase = require("./client");
 
 const Schema = mongoose.Schema;
 
+var address = {
+  addressName: { type: String },
+  name: { type: String },
+  addressLine: { type: String },
+  city: { type: String },
+  state: { type: String },
+  zipCode: { type: String },
+  phone: { type: String },
+};
+
+var shoppingList = {
+  title: { type: String },
+  wishedProducts: [],
+};
+
+var shoppingCart = {
+  productsIn: { type: String },
```

Comment:
Shopping cart should be a list of product and product amount tuple.

Reply:
Added as: 
`productsIn: [({ type: Schema.Types.ObjectId }, { type: Number })], // productId and productAmount
`